pinot-core/src/main/java/org/apache/pinot/core/indexsegment/generator/SegmentGeneratorConfig.java

```diff
@@ -190,13 +190,17 @@ public SegmentGeneratorConfig(@Nullable TableConfig tableConfig, Schema schema)
     _segmentPartitionConfig = indexingConfig.getSegmentPartitionConfig();
 
     // Star-tree V2 configs
-    List<StarTreeIndexConfig> starTreeIndexConfigs = indexingConfig.getStarTreeIndexConfigs();
-    if (starTreeIndexConfigs != null && !starTreeIndexConfigs.isEmpty()) {
-      List<StarTreeV2BuilderConfig> starTreeV2BuilderConfigs = new ArrayList<>(starTreeIndexConfigs.size());
-      for (StarTreeIndexConfig starTreeIndexConfig : starTreeIndexConfigs) {
-        starTreeV2BuilderConfigs.add(StarTreeV2BuilderConfig.fromIndexConfig(starTreeIndexConfig));
+    if (indexingConfig.isEnableDefaultStarTree()) {
```

Comment:
If both are defined, you should pay attention to the specific start tree definition requested. This will keep backward compat also.

Reply:
This won't have backward compatible issue because default star-tree is not enabled by default.
But this is a great question, and we should not ignore the explicitly configured star-tree. Modify the code to generate star-tree for both configs (if both are defined, generate both default and customized).